src/app/content/components/ContentLink.tsx

```diff
@@ -53,7 +54,23 @@ export const ContentLink = (props: React.PropsWithChildren<Props>) => {
     hasUnsavedHighlight,
     ...anchorProps
   } = props;
-  const {url, params} = getBookPageUrlAndParams(book, page);
+  const services = useServices();
+  const [url, setUrl] = React.useState('');
+  const [params, setParams] = React.useState<Params>({} as any);
+
+  React.useEffect(() => {
+    const setBookPageUrlAndParams = async() => {
```

Comment:
all the same comments as the other place

Reply:
We could create `useGetBookPageUrlAndParams` hook that will return `{isLoading: boolean, url: string, params: ...}` and reuse it